code/game/gamemodes/wizard/artefact.dm

```diff
@@ -819,3 +819,38 @@ GLOBAL_LIST_EMPTY(multiverse)
 	heal_brute = 25
 	heal_burn = 25
 	heal_oxy = 25
+
+/obj/item/reagent_containers/food/drinks/everfull
+	name = "everfull mug"
+	desc = "An enchanted mug which can be filled with any of various liquids on command."
+	icon_state = "evermug"
+
+/obj/item/reagent_containers/food/drinks/everfull/attack_self(mob/user)
+	var/static/list/options = list("Omnizine" = image(icon = 'icons/obj/storage.dmi', icon_state = "firstaid"),/// todo add icons for these
+							"Ale" = image(icon = 'icons/obj/drinks.dmi', icon_state = "alebottle"),
+							"Wine" = image(icon = 'icons/obj/drinks.dmi', icon_state = "wineglass"),
+							"Holy Water" = image(icon = 'icons/obj/drinks.dmi', icon_state = "holyflask"),
+							"Welder Fuel" = image(icon = 'icons/obj/objects.dmi', icon_state = "fuel"),
+							"Vomit" = image(icon = 'icons/effects/blood.dmi', icon_state = "vomit_1"),)
+	var/static/list/options_to_reagent = list("Omnizine" = "omnizine",
+									"Ale" = "ale",
+									"Wine" = "wine",
+									"Holy Water" = "holywater",
+									"Welder Fuel" = "fuel",
+									"Vomit" = "vomit",)
+	var/static/list/options_to_descriptions = list("Omnizine" = "a strange pink-white liquid",
+												"Ale" = "foamy amber ale",
+												"Wine" = "deep red wine",
+												"Holy Water" = "sparkling clear water",
+												"Welder Fuel" = "a dark, pungent, oily substance",
+												"Vomit" = "warm chunky vomit",)
```

Comment:
```suggestion
												"Vomit" = "warm chunky vomit")
```
End of lists dont need that

Reply:
I copy/pasted the format from how spellblade does it, and I was afraid to change that bit : P
I'll remove that from the lists then.